common/src/main/kotlin/org/valkyrienskies/mod/util/RelocationUtil.kt

```diff
@@ -48,70 +47,60 @@ fun relocateBlock(
         tag
     }
 
-    rotateBlockState(state, direction)
+    state = state.rotate(rotation)
 
     val level = toChunk.level
 
     fromChunk.setBlockState(from, AIR, false)
-    level.sendBlockUpdated(from, state, AIR, 0)
     toChunk.setBlockState(to, state, false)
-    level.sendBlockUpdated(to, state, AIR, 0)
-    level.chunkSource.lightEngine.checkBlock(from)
-    level.chunkSource.lightEngine.checkBlock(to)
+
+    if (doUpdate) {
+        updateBlock(level, from, to, state)
+    }
 
     tag?.let {
-        val be = toChunk.getBlockEntity(to, CHECK)!!
-        if (be is ShipBlockEntity)
-            be.ship = toShip
+        val be = level.getBlockEntity(to)!!
+        if (be is ShipBlockEntity) be.ship = toShip
 
         be.load(state, it)
     }
 }
 
-private fun addDirection(direction1: Direction, direction2: Direction) =
-    Direction.from2DDataValue((direction1.get2DDataValue() + direction2.get2DDataValue()) and 3)
-
-private fun rotateBlockState(state: BlockState, direction: Direction) {
-    if (direction == NORTH) return
-    // TODO there are prob more relevant states that need to get modified
-    if (state.hasProperty(BlockStateProperties.HORIZONTAL_FACING)) {
-        state.setValue(
-            BlockStateProperties.HORIZONTAL_FACING,
-            addDirection(state.getValue(BlockStateProperties.HORIZONTAL_FACING), direction)
-        )
-    } else if (state.hasProperty(BlockStateProperties.FACING)) {
-        state.setValue(
-            BlockStateProperties.FACING,
-            addDirection(state.getValue(BlockStateProperties.FACING), direction)
-        )
-    } else if (state.hasProperty(BlockStateProperties.AXIS)) {
-        state.setValue(
-            BlockStateProperties.AXIS,
-            if (direction.axis == Direction.Axis.X)
-                if (state.getValue(BlockStateProperties.AXIS) == Direction.Axis.X)
-                    Direction.Axis.Z
-                else if (state.getValue(BlockStateProperties.AXIS) == Direction.Axis.Z)
-                    Direction.Axis.X
-                else
-                    Direction.Axis.Y
-            else
-                state.getValue(BlockStateProperties.AXIS)
-        )
-    } else if (state.hasProperty(BlockStateProperties.FACING_HOPPER)) {
-        state.setValue(
-            BlockStateProperties.FACING,
-            addDirection(state.getValue(BlockStateProperties.FACING), direction)
-        )
-    }
+/**
+ * Update block after relocate
+ *
+ * @param level
+ * @param fromPos old position coordinate
+ * @param toPos new position coordinate
+ * @param toState new blockstate at toPos
+ */
+fun updateBlock(level: Level, fromPos: BlockPos, toPos: BlockPos, toState: BlockState) {
+
+    level.setBlocksDirty(fromPos, toState, AIR)
+    level.sendBlockUpdated(fromPos, toState, AIR, 75)
+    //This handles the update for neighboring blocks in worldspace
+    AIR.updateNeighbourShapes(level, fromPos, 75 and -0x22, 512 - 1)
```

Comment:
what the hell is going on here?! Please add a comment that explains from where u got these numbers and wherefor they are used.

Reply:
Documented. Not sure where the -0x22 came from, but it was unnecessary, and so was removed.